cql3/restrictions/statement_restrictions.cc

```diff
@@ -326,15 +326,16 @@ static std::vector<expr::expression> extract_clustering_prefix_restrictions(
 statement_restrictions::statement_restrictions(data_dictionary::database db,
         schema_ptr schema,
         statements::statement_type type,
-        const std::vector<expr::expression>& where_clause,
+        const expr::expression& where_clause,
         prepare_context& ctx,
         bool selects_only_static_columns,
         bool for_view,
         bool allow_filtering)
     : statement_restrictions(schema, allow_filtering)
 {
-    if (!where_clause.empty()) {
-        for (auto&& relation_expr : where_clause) {
+    auto terms = boolean_factors(where_clause);
+    if (!terms.empty()) {
```

Comment:
This `if` sounds superfluous, iterating over an empty collection is well defined.

Reply:
It's just replacing the same `if` we had before. I'll add a patch removing it.